src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -0,0 +1,315 @@
+package io.libp2p.security.noise
+
+import com.google.protobuf.ByteString
+import com.southernstorm.noise.protocol.CipherState
+import com.southernstorm.noise.protocol.CipherStatePair
+import com.southernstorm.noise.protocol.DHState
+import com.southernstorm.noise.protocol.HandshakeState
+import com.southernstorm.noise.protocol.Noise
+import io.libp2p.core.P2PAbstractChannel
+import io.libp2p.core.PeerId
+import io.libp2p.core.crypto.PrivKey
+import io.libp2p.core.crypto.marshalPublicKey
+import io.libp2p.core.crypto.unmarshalPublicKey
+import io.libp2p.core.multistream.Mode
+import io.libp2p.core.multistream.ProtocolMatcher
+import io.libp2p.core.security.SecureChannel
+import io.libp2p.etc.SECURE_SESSION
+import io.libp2p.etc.events.SecureChannelFailed
+import io.libp2p.etc.events.SecureChannelInitialized
+import io.libp2p.etc.types.toByteArray
+import io.libp2p.etc.types.toByteBuf
+import io.netty.buffer.ByteBuf
+import io.netty.channel.ChannelHandlerContext
+import io.netty.channel.ChannelInboundHandlerAdapter
+import io.netty.channel.SimpleChannelInboundHandler
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.apache.logging.log4j.core.config.Configurator
+import spipe.pb.Spipe
+import java.util.Arrays
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.atomic.AtomicInteger
+
+class NoiseXXSecureChannel(private val localKey: PrivKey) :
+        SecureChannel {
+
+    companion object {
+        const val protocolName = "Noise_XX_25519_ChaChaPoly_SHA256"
+        const val announce = "/noise/$protocolName/0.1.0"
+
+        @JvmStatic
+        var localStaticPrivateKey25519: ByteArray = ByteArray(32)
+
+        init {
+            // initialize static Noise key
+            Noise.random(localStaticPrivateKey25519)
+        }
+    }
+
+    private var logger: Logger
+    private var loggerNameParent: String = NoiseXXSecureChannel::class.java.name + this.hashCode()
+    private lateinit var chid: String
+
+    private lateinit var role: AtomicInteger
+
+    private val handshakeHandlerName = "NoiseHandshake"
+
+    override val announce = Companion.announce
+    override val matcher = ProtocolMatcher(Mode.PREFIX, name = "/noise/$protocolName/0.1.0")
+
+    init {
+        logger = LogManager.getLogger(loggerNameParent)
+        Configurator.setLevel(loggerNameParent, Level.DEBUG)
```

Comment:
I would not use programmatic logging configuration unless it's unavoidable. 
Why `DEBUG` level is hardcoded here?

Reply:
Done.